packages/grid/data-grid/src/index.ts

```diff
@@ -2,6 +2,3 @@ export * from '../../_modules_';
 export * from './DataGrid';
 export * from './DataGridProps';
 export * from './useDataGridComponent';
-// TODO v5, remove this should be handled userland
-// https://github.com/mui-org/material-ui/pull/21279#discussion_r435748054
-export * from './themeAugmentation';
```

Comment:
AFAIK, we can't remove it yet. Develpers need to be able to import deep `@mui/x-data-grid/themeAugmentation `